blog/02-15-2023-build-feature-rich-persistent-dapp-on-celo-using-wagmi/build-feature-rich-persistent-dapp-on-celo-using-wagmi.md

```diff
@@ -0,0 +1,747 @@
+
+
+## Frontend
+
+Exit the current directory into the root folder and install the dependencies: 
+
+```bash
+cd ..
+cd frontend
+yarn install
+```
+
+Since I already completed the setup, it's important that I show you how to set up the frontend folder.
+
+If we are to create a new project from the scratch, let's assume we're in the root directory with the `backend` being the only folder. The recommended way to start a wagmi project is to run the following command.
+
+```bash
+npm init wagmi -- --template next-connectkit
+```
+Select the choice of provider.
+
+![image](images/2.png)
+
+Installation success
+
+![image](images/6.png)
+
+The command takes extra arguments `--template` and `next-connectkit` to start the project with a boilerplate that uses connectkit compatible with NextJs to display information to users on the frontend.
+
+
+Follow the instructions to make your choice. For this project, I have selected the 'default' option which uses 'NextJs'. You really do not need to do anything else. Much of the work is done for you in the background.
+
+Before we proceed to set up wagmi, let's understand what Wagmi aims to achieve.
+
+**What is Wagmi?**
+Wagmi is a library for interacting with Ethereum-based applications. It comprises over 20 functional React hooks to perform almost any actions that a typical dapp needs.
+
+<!-- >Note: It is recommended to set up  -->
+
+- `wagmi.ts`
+
+For wagmi to work effectively, we need to use it as a provider and top-level tag for the application or the part where we need it. The provider requires a client to be parsed as a property. So first, we will set up a configuration to get the client ready for import. Below we set the value of `getDefaultClient()` from 'connectKit' as the provider value for the `createClient` function imported from 'wagmi'. The result is the client we will feed to the wagmi provider component. It works more like a plugin. The 'getDefaultClient' function internally sets up known connectors for us. So we do not need to explicitly configure providers like Metamask, Coinbase, and Walletconnect. This is one of the advantages of using wagmi over other libraries such as web3React. 
+
+Lastly, we get the already configured `celoAlfajores` network information from the `'wagmi/chains'` module. Now, the work is a lot easier. 
+
+```ts
+import { getDefaultClient } from 'connectkit'
+import { createClient } from 'wagmi'
+import { celoAlfajores } from 'wagmi/chains'
+
+export const client = createClient(
+  getDefaultClient({
+    autoConnect: true,
+    appName: 'DRythm Music-sharing dapp',
+    chains: [celoAlfajores]
+  })
+)
+```
+
+- `_app.ts`
+
+Import the client from the previous step in 'wagmi.ts'.
+Import WagmiConfig provider from 'wagmi'. We'll need this to wrap and initialize our dapp so wagmi can effectively manage it for us. It will give us access to the various wagmi hooks in the child components.
```

Comment:
Import WagmiConfig provider from 'wagmi'. We'll need this to wrap and initialize our Dapp so wagmi can effectively manage it. It will give us access to the various wagmi hooks in the child components.